Beyla crashes `panic: runtime error: index out of range [1] with length 1`

We install Beyla using the Helm chart in a GKE cluster.

One of the Beyla pods in the daemonset continuously crashes with the following exception:

```
time=2024-10-11T03:47:25.820Z level=INFO msg="new process for already instrumented executable" component=discover.TraceAttacher pid=15718 child=[] exec=/usr/local/bin/envoy
time=2024-10-11T03:47:25.820Z level=INFO msg="new process for already instrumented executable" component=discover.TraceAttacher pid=3912485 child=[] exec=/usr/local/bin/pilot-agent
panic: runtime error: index out of range [1] with length 1

goroutine 257 [running]:
encoding/binary.bigEndian.Uint16(...)
    /usr/local/go/src/encoding/binary/binary.go:146
github.com/grafana/beyla/pkg/internal/ebpf/common.getTopicNameSize({0xc00272eeb8?, 0xc00272eeb8?, 0x410ae5?}, 0x54?, 0x54?, 0x0?)
    /opt/app-root/pkg/internal/ebpf/common/kafka_detect_transform.go:257 +0xfb
github.com/grafana/beyla/pkg/internal/ebpf/common.getTopicName({0xc00272eeb8, 0x80, 0x80}, 0x7b, 0x1, 0xa)
    /opt/app-root/pkg/internal/ebpf/common/kafka_detect_transform.go:212 +0x5f
github.com/grafana/beyla/pkg/internal/ebpf/common.ProcessKafkaRequest({0xc00272eeb8, 0x80, 0x80})
    /opt/app-root/pkg/internal/ebpf/common/kafka_detect_transform.go:93 +0x106
github.com/grafana/beyla/pkg/internal/ebpf/common.ProcessPossibleKafkaEvent(0xc00272ed80, {0xc00272edb8?, 0x80?, 0x4009e40?}, {0xc00272eeb8, 0x80, 0x80})
    /opt/app-root/pkg/internal/ebpf/common/kafka_detect_transform.go:59 +0x52
github.com/grafana/beyla/pkg/internal/ebpf/common.ReadTCPRequestIntoSpan(_, {_, _})
    /opt/app-root/pkg/internal/ebpf/common/tcp_detect_transform.go:68 +0x5e7
github.com/grafana/beyla/pkg/internal/ebpf/common.ReadBPFTraceAsSpan(_, {_, _})
    /opt/app-root/pkg/internal/ebpf/common/common.go:124 +0x330
github.com/grafana/beyla/pkg/internal/ebpf/common.(*ringBufForwarder).processAndForward(0xc000b28120, {{0xc0039c1400, 0x200, 0x200}, 0x208}, 0xc0004791f0)
    /opt/app-root/pkg/internal/ebpf/common/ringbuf.go:173 +0x16b
github.com/grafana/beyla/pkg/internal/ebpf/common.(*ringBufForwarder).readAndForwardInner(0xc000b28120, {0x2796168, 0xc0004c4460}, 0xc0004791f0)
    /opt/app-root/pkg/internal/ebpf/common/ringbuf.go:159 +0x159
github.com/grafana/beyla/pkg/internal/ebpf/common.(*ringBufForwarder).sharedReadAndForward(0xc000b28120, {0x27b07f8, 0xc0000ba730}, {0xc001a6b408, 0x242, 0x24f}, 0xc0004791f0)
    /opt/app-root/pkg/internal/ebpf/common/ringbuf.go:107 +0x23c
github.com/grafana/beyla/pkg/internal/ebpf/nethttp.(*Tracer).Run(0xc000bde000, {0x27b07f8, 0xc0000ba730}, 0xc0004791f0)
    /opt/app-root/pkg/internal/ebpf/nethttp/nethttp.go:251 +0xfa
created by github.com/grafana/beyla/pkg/internal/ebpf.(*ProcessTracer).Run in goroutine 183
    /opt/app-root/pkg/internal/ebpf/tracer_linux.go:36 +0x233
```

Beyla image version: 1.8.5

Please kindly advise!

--- a/pkg/internal/ebpf/common/kafka_detect_transform.go
+++ b/pkg/internal/ebpf/common/kafka_detect_transform.go
@@ -253,7 +253,7 @@ func getTopicNameSize(pkt []byte, offset int, op Operation, apiVersion int16) (i
 		if err != nil {
 			return 0, err
 		}
-	} else if offset < len(pkt) {
+	} else if offset < (len(pkt) - 1) { // we need at least 2 bytes to read uint16
 		topicNameSize = int(binary.BigEndian.Uint16(pkt[offset:]))
 	}
 	if topicNameSize <= 0 {

--- a/pkg/internal/ebpf/common/kafka_detect_transform_test.go
+++ b/pkg/internal/ebpf/common/kafka_detect_transform_test.go
@@ -14,6 +14,15 @@ func TestProcessKafkaRequest(t *testing.T) {
 		expected *KafkaInfo
 	}{
 		{
+			name:  "Fetch request (v11) truncated - 1",
+			input: []byte{0, 0, 0, 94, 0, 1, 0, 11, 0, 0, 0, 224, 0, 6, 115, 97, 114, 97, 109, 97, 255, 255, 255, 255, 0, 0, 1, 244, 0, 0, 0, 1, 6, 64, 0, 0, 0, 0, 0, 0, 0, 255, 255, 255, 255, 0, 0, 0, 1, 0},
+			expected: &KafkaInfo{
+				ClientID:    "sarama",
+				Operation:   Fetch,
+				TopicOffset: 45,
+			},
+		},
+		{
 			name:  "Fetch request (v11) truncated",
 			input: []byte{0, 0, 0, 94, 0, 1, 0, 11, 0, 0, 0, 224, 0, 6, 115, 97, 114, 97, 109, 97, 255, 255, 255, 255, 0, 0, 1, 244, 0, 0, 0, 1, 6, 64, 0, 0, 0, 0, 0, 0, 0, 255, 255, 255, 255, 0, 0, 0, 1},
 			expected: &KafkaInfo{

